metapool/sample_sheet.py

```diff
@@ -1983,73 +2067,86 @@ def _id_sample_sheet_class_from_dict(dict_w_header_info):
 
 
 def _id_sample_sheet_class(sheet_type, sheet_version, assay_type):
-    def _make_version_err_msg(sheet_type, sheet_version):
-        return f"'{sheet_version}' is an unrecognized SheetVersion for " \
-               f"'{sheet_type}'"
-
-    def _make_assay_err_msg(assay_type):
-        return f"'{assay_type}' is an unrecognized Assay type"
-
-    def check_metag_assay_type(assay_type):
-        if assay_type != _METAGENOMIC:
-            raise ValueError(_make_assay_err_msg(assay_type))
-
-    if sheet_type == STANDARD_METAG_SHEET_TYPE:
-        if assay_type == _METAGENOMIC:
-            if sheet_version == '102':
-                sheet_class = MetagenomicSampleSheetv102
-            elif sheet_version == '101':
-                sheet_class = MetagenomicSampleSheetv101
-            elif sheet_version == '90':
-                sheet_class = MetagenomicSampleSheetv90
-            elif sheet_version in ['95', '99', '100']:
-                # 95, 99, and v100 are functionally the same type.
-                sheet_class = MetagenomicSampleSheetv100
-            else:
-                raise ValueError(
-                    _make_version_err_msg(sheet_type, sheet_version))
-        elif assay_type == _METATRANSCRIPTOMIC:
-            sheet_class = MetatranscriptomicSampleSheetv0
-        else:
-            raise ValueError(_make_assay_err_msg(assay_type))
-    elif sheet_type == STANDARD_METAT_SHEET_TYPE:
-        if assay_type == _METATRANSCRIPTOMIC:
-            if sheet_version == '0':
-                sheet_class = MetatranscriptomicSampleSheetv0
-            elif sheet_version == '10':
-                sheet_class = MetatranscriptomicSampleSheetv10
+    SAMPLE_SHEETS_BY_SHEET_TYPE = {
+        STANDARD_METAG_SHEET_TYPE: {
+            _METAGENOMIC: {
+                '102': MetagenomicSampleSheetv102,
+                '101': MetagenomicSampleSheetv101,
+                '100': MetagenomicSampleSheetv100,
+                '99': MetagenomicSampleSheetv100,
+                '95': MetagenomicSampleSheetv100,
+                '90': MetagenomicSampleSheetv90,
+            },
+            # NB: this really shouldn't be here under metag, but
+            # things were a little confused in the early days ...
+            _METATRANSCRIPTOMIC: {
```

Comment:
This feels like can cause problems but I guess it doesn't ...

Reply:
Well, what WILL cause problems is continuing to let MetatranscriptomicSampleSheetv0 be selected in two branches--our validation will not longer allow this, because there is now only one right way to define the Assay and SheetType values for a MetatranscriptomicSampleSheetv0 object.  I think it makes more sense to remove this definition of a MetatranscriptomicSampleSheetv10 sheet (that has "standard_metag" as its sheet type) than it does to remove the definition that is actually right, so I'm going to do that.  However, this may mean we have to change some older files if/when we want to run them through again.